### ISSUE TITLE
Pursue FSF endorsement? 

### ISSUE THREAD

**Issue by SolidHal** [SolidHal/Librean#28]
Once all major bugs are fixed and the package repo is up, pursue this checklist https://libreplanet.org/wiki/Template:FSDG_Checklist

**Comment by SolidHal** [comment]
Idea: Instead of copying all of debians repos, add a package that conflicts with all non-free packages, akin to "your freedom"

**Comment by jonassmedegaard** [comment]
You most likely won't achieve FSF endorsement unless distributing independent of debian.org domain: Reason Debian itself is not endorsed is that non-free stuff _exist_ on same debian.org domain, even though it requires explicit action from the user to make any use of it (much like your idea above).

**Comment by SolidHal** [comment]
Yeah, I've been chatting with someone interning at the FSF about just that. It'd be nice to leverage debians large set of mirrors, and not have to duplicate all of their infrastructure though :/

**Comment by endian-albin** [comment]
I'd be willing to provide a Swedish mirror for PrawnOS if the required amount of storage isn't too much. How much do you think is needed per arch?

**Comment by albsod** [comment]
I'd be willing to provide a Swedish mirror for PrawnOS if the required amount of storage isn't too much. How much do you think is needed per arch?

**Comment by SolidHal** [comment]
@albsod likely 100s of gigabytes to mirror all of the packages from each arch would be required. The next biggest thing holding this back would be firefox, distros with firefox can't get FSF endorsement.

**Comment by albsod** [comment]
Oh, that's a lot. Is there a way to find out the total size of just the compressed armhf packages in main?

**Comment by gregordinary** [comment]
The armhf archive size is 275 GB:
https://www.debian.org/mirror/size

It'll need a bit more coordination to confirm we have resources, but I'd be willing to contribute some monthly amount for hosting to help achieve this.

**Comment by SolidHal** [comment]
Thanks @gregordinary, thats actually less than I figured. Creating a PrawnOS mirror wouldn't be too too expensive from a storage side, but the bandwidth will likely be the pricey bit. I'm not familiar with how to estimate how much bandwith will be required so I'll have to spend some time figuring that out. 

Please share any opinions on Firefox replacements, and if FSF endorsement is "worth" exchanging Firefox for Icecat/Iceweasel/something else.

**Comment by austin987** [comment]
IMO it would be nice to have, but I'd be concerned that they might object to blobby_kernel support being available, which I'd like to see remain. If they're okay with that, then I'm onboard. Not hard for someone to install firefox on their own if they really want to.

**Comment by herbsmn** [comment]
Not sure if this is helpful or not but PureBrowser is the Firefox fork that PureOS, a FSF FSDG certified OS, uses. https://software.pureos.net/package/bin/amber/purebrowser

**Comment by SolidHal** [comment]
@austin987 blobby_kernel support *should* be fine, distributing images built with a blobby kernel might be a different story. Not sure though. 
blobby_kernel support is useful as a debug too.

**Comment by jonassmedegaard** [comment]
PureBrowser is no longer maintained: PureOS now ships with regular Debian-maintained Firefox ESR.

**Comment by SolidHal** [comment]
@jonassmedegaard do you know how PureOS manages to get FSF endorsed while using standard Firefox ESR?

**Comment by jonassmedegaard** [comment]
Quoting SolidHal (2020-12-19 18:01:10)
> @jonassmedegaard do you know how PureOS manages to get FSF endorsed while using standard Firefox ESR? 

Only FSF can answer that question: Neither me nor PureOS nor fans of FSF 
decides what FSF chooses to endorse.

I was previously hired by Purism to maintain a patched Firefox ESR 
rebranded as PureBrowser.

I still do work for Purism, but doing other tasks now (e.g. packaging 
web-apps for the matrix.org federated messaging network).


 - Jonas

-- 
 * Jonas Smedegaard - idealist & Internet-arkitekt
 * Tlf.: +45 40843136  Website: http://dr.jones.dk/

 [x] quote me freely  [ ] ask before reusing  [ ] keep private

**Comment by raghavgururajan** [comment]
Mozilla Firefox can be replaced with [GNU Icecat](https://www.gnu.org/software/gnuzilla/).

Google-Chromium can be replaced with [Ungoogled-Chromium](https://github.com/Eloston/ungoogled-chromium).

**Comment by ncorder** [comment]
@solidhat I ran a small organization called the EPuters Project.

We would most likely be able to aid in basic hosting of repositories ( up to about 500GB ) until your project matures more as long as you are seeking FSDG compliance.

Let me know if that would interest you, you can email me at admin@eputers.net .
I would also highly advise the project move away from Github to [NotABug](https://notabug.org/) or another platform.
